scripts/NewznabToCardigannYML.ps1

```diff
@@ -7,62 +9,329 @@
+                optional = $true
+            }
+            genre       = [ordered]@{
+                selector = 'genre'
+                optional = $true
+            }
+            year        = [ordered]@{
+                selector = 'year'
+                optional = $true
+            }
+            category    = [ordered]@{
+                selector  = 'attr[name="category"]:nth-of-type(1)'
+                attribute = 'value'
+            }
+            category2   = [ordered]@{
+                optional  = $true
+                selector  = 'attr[name="category"]:nth-of-type(2)'
+                attribute = 'value'
+            }
+            size        = [ordered]@{
+                selector  = 'attr[name="size"]'
+                attribute = 'value'
+            }
+            grabs       = [ordered]@{
+                selector  = 'attr[name="grabs"]'
+                attribute = 'value'
+            }
+            poster      = [ordered]@{
+                selector  = 'attr[name="poster"]'
+                attribute = 'value'
+            }
+            files       = [ordered]@{
+                selector  = 'attr[name="files"]'
+                attribute = 'value'
+                optional  = $true
+            }
+            coverurl    = [ordered]@{
+                selector  = 'attr[name="coverurl"]'
+                attribute = 'value'
+            }
+        }
+    }
+    download              = [ordered]@{
+        error = @(
+            @(
+                [ordered]@{
+                    selector = 'error'
+                    message  = [ordered]@{
+                        selector  = 'error'
+                        attribute = 'description'
+                    }
+                }
+            )
+        )
+    }
+}
+Write-Information 'YML Built'
 
-# ToDo: Validate Categories List (Names) - use newznabcats.txt
-# None matching categories (case insensitive) will need to be commented out
+$ymlout = '---
+'
+$ymlout += (Invoke-YMLReplace -Name $($hashTable | ConvertTo-Yaml))
+$ymlout = ($ymlout).replace("'[", '[')
+$ymlout = ($ymlout).replace("]'", ']')
+$ymlout = ((($ymlout) -replace '\\\\', '\') -replace '---', '---').Trim()
+$ymlout += '
+# newznab standard'
+# return $ymlout
 
-# ToDo: Create YML file from given data inputs
+Write-Information 'Indexer YML Complete'
+$ymlout | Out-File $OutputFile -Encoding utf8
+$categoryCsv = Import-Csv $($PSScriptRoot + [system.Io.Path]::DirectorySeparatorChar + 'newznabcats(v4.0).csv')
+[System.Collections.Generic.List[string]]$linesToReplace = @()
+$ymlout = Get-Content $OutputFile -Encoding utf8
+#Validating Category Names
+foreach ($line in ($ymlout | Select-String '{ id:' | Select-Object -ExpandProperty Line))
+{
+    foreach ($cleanLine in ($line -replace '- { ', '' -replace ' }' -replace 'id: ', '' -replace 'cat: ').Trim())
+    {
+        $split = $cleanLine -split ', '
+        $categoryCsvCategory = ($categoryCsv | Where-Object { $_.id -eq $split[0] }).newznabcat
```

Comment:
we cannot use ID to match.  one of the big purposes of switching to YML is to allow remapping of a site's cat # to the correct newznab cat as they do not follow standards